pydocx/export/base.py

```diff
@@ -199,26 +201,47 @@ def __init__(
     def parse_run_properties(self, el, parsed, stack):
         properties = RunProperties.load(el)
         parent = stack[-1]['element']
-        self.document.main_document_part.style_definitions_part.save_properties_for_element(parent, properties)  # noqa
+        self.style_definitions_part.save_properties_for_element(parent, properties)  # noqa
 
     def parse_paragraph_properties(self, el, parsed, stack):
         properties = ParagraphProperties.load(el)
         parent = stack[-1]['element']
-        self.document.main_document_part.style_definitions_part.save_properties_for_element(parent, properties)  # noqa
+        self.style_definitions_part.save_properties_for_element(parent, properties)  # noqa
+
+    @property
+    def document(self):
+        if self._document:
+            return self._document
+        return self.load_document()
```

Comment:
To avoid the problem where `load_document` in a subclass forgets to assign to `self._document`, what about doing that here, also?